server/license: Logging and error message cleanup

spilchen opened the pull request:
This addresses a few late comments from PR #130872:
- The license expiry date is now included when throttling due to an expired license.
- Additional context from the license enforcer is logged whenever a license change occurs.
- The throttle message has been clarified to explicitly state that throttling is due to reaching the maximum number of concurrent open transactions.

Epic: CRDB-39988
Informs: CRDB-39991
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131022)
<!-- Reviewable:end -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131022#-:-O7VOl9832_7eNGEDGMN:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/server/license/enforcer.go` line 382 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131022#-O7VOgnM4x4RnNbD_tU3:-O7VOgnM4x4RnNbD_tU4:b-h8hnkw) ([raw file](https://github.com/cockroachdb/cockroach/blob/d49f8a0952e4f9b6191729b6d6c2bf801427310b/pkg/server/license/enforcer.go#L382)):*
> ```Go
> 	expiry := timeutil.Unix(e.licenseExpiryTS.Load(), 0)
> 	if !expiry.IsZero() {
> 		sb.WriteString(fmt.Sprintf("expiration at %s, ", expiry))
> ```

Nit: Do we wanted this quoted like the grace period above.


<!-- Sent from Reviewable.io -->

spilchen committed:
server/license: Logging and error message cleanup

This addresses a few late comments from PR #130872:
- The license expiry date is now included when throttling due to an expired
  license.
- Additional context from the license enforcer is logged whenever a license
  change occurs.
- The throttle message has been clarified to explicitly state that throttling
  is due to reaching the maximum number of concurrent open transactions.

Epic: CRDB-39988
Informs: CRDB-39991
Release note: None

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131022)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @fqazi)

___
*[`pkg/server/license/enforcer.go` line 382 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131022#-O7VOgnM4x4RnNbD_tU3:-O7VTZ9J4h7_vaoHu6NV:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/d49f8a0952e4f9b6191729b6d6c2bf801427310b/pkg/server/license/enforcer.go#L382)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Nit: Do we wanted this quoted like the grace period above.
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR @fqazi 

bors r=fqazi
